wallet-web/components/delegate/DelegateForm.tsx

```diff
@@ -1,121 +1,146 @@
-import Grid from "@material-ui/core/Grid";
-import React from "react";
-import { Button, InputAdornment } from "@material-ui/core";
-import TextField from "@material-ui/core/TextField";
-import { DENOM } from "../../pages/_app";
-import { theme } from "../../lib/theme";
-import { basicRawCoinValueValidation, makeBasicStyle, validateIdentityKey } from "../../common/helpers";
-
+import React, { useState, useEffect } from 'react'
+import Grid from '@material-ui/core/Grid'
+import { Button, InputAdornment } from '@material-ui/core'
+import TextField from '@material-ui/core/TextField'
+import { Alert } from '@material-ui/lab'
+import { DENOM } from '../../pages/_app'
+import { theme } from '../../lib/theme'
+import {
+  basicRawCoinValueValidation,
+  makeBasicStyle,
+  validateIdentityKey,
+} from '../../common/helpers'
+import { useGetBalance } from '../../hooks/useGetBalance'
 
 type DelegateFormProps = {
-    onSubmit: (event: any) => void
+  onSubmit: (event: any) => void
 }
 
 export default function DelegateForm(props: DelegateFormProps) {
-    const classes = makeBasicStyle(theme);
-
-    const [validAmount, setValidAmount] = React.useState(true)
-    const [validIdentity, setValidIdentity] = React.useState(true)
-    // const [checkboxSet, setCheckboxSet] = React.useState(false)
-
-    // const handleCheckboxToggle = () => {
-    //     setCheckboxSet((prevSet) => !prevSet);
-    // }
-
-    const handleAmountChange = (event: any) => {
-        let nonZeroAmount = event.target.value.length > 0
-        if (nonZeroAmount) {
-            // don't ask me about that. javascript works in mysterious ways
-            // and this is apparently a good way of checking if string
-            // is purely made of numeric characters
-            let parsed = +event.target.value
-            if (isNaN(parsed)) {
-                setValidAmount(false)
-            } else {
-                setValidAmount(true)
-            }
-        }
+  const classes = makeBasicStyle(theme)
+
+  const [validAmount, setValidAmount] = useState(true)
+  const [validIdentity, setValidIdentity] = useState(true)
+  const [allocationWarning, setAllocationWarning] = useState(false)
+  const { getBalance, accountBalance } = useGetBalance()
+
+  useEffect(() => {
+    getBalance()
+  }, [getBalance])
+
+  // const [checkboxSet, setCheckboxSet] = React.useState(false)
+
+  // const handleCheckboxToggle = () => {
+  //     setCheckboxSet((prevSet) => !prevSet);
+  // }
+
+  const handleAmountChange = (event: any) => {
+    // don't ask me about that. javascript works in mysterious ways
+    // and this is apparently a good way of checking if string
+    // is purely made of numeric characters
+    let parsed = +event.target.value
+    if (isNaN(parsed)) {
+      setValidAmount(false)
+    } else {
+      if (parsed > 0 && parseInt(accountBalance.amount) - parsed < 1) {
```

Comment:
Could we change that to instead show the warning if after delegating user is going to have less the gas fee? (so that he would be able to undelegate in the future)

Reply:
Yep. That's done. It's part of a helper function now.